src/kw/kw.cpp

```diff
@@ -259,7 +261,7 @@ bool KanoWorld::is_logged_in(const bool verbose)
         return false;
     }
 
-    std::time_t duration = atol(this->expiration_date.c_str());
+    std::time_t duration = atol(this->get_expiration_date().c_str());
```

Comment:
Should this be wrapped in a `try-catch`?

Reply:
`atol` [guarantees](http://www.cplusplus.com/reference/cstdlib/atol/) that it won't throw an exception and `get_expiration_date()` doesn't either so not necessary.